src/ServiceManagement/RecoveryServices/Commands.RecoveryServices/Service/InvokeAzureRecoveryServicesVaultUpgrade.cs

```diff
@@ -0,0 +1,238 @@
+                record.PercentComplete = (int)elapsedSeconds * 100 / (int)taskTimeoutInSeconds;
+                this.WriteProgress(record);
+            }
+            while (response.OperationStatus == Constants.InProgress &&
+                elapsedSeconds < taskTimeoutInSeconds &&
+                !this.StopProcessingFlag);
+
+            this.WriteResponse(
+                string.Format(
+                    Properties.Resources.TrackVaultUpgradeJobResult,
+                    response.OperationResult,
+                    response.OperationStatus));
+
+            if (response.OperationStatus == Constants.Completed)
+            {
+                if (response.OperationResult == Constants.Succeeded)
+                {
+                    this.WriteResponse(
+                        string.Format(
+                            Properties.Resources.VaultUpgradeSucceded,
+                            this.ResourceType,
+                            this.VaultName));
+                }
+                else
+                {
+                    this.WriteResponse(Properties.Resources.VaultUpgradeNotSucceded);
+                }
+            }
+            else if (this.StopProcessingFlag)
+            {
+                this.WriteResponse(Properties.Resources.VaultUpgradeTerminated);
+            }
+            else
+            {
+                this.WriteResponse(Properties.Resources.VaultUpgradeTimedOut);
+            }
+        }
+
+        /// <summary>
+        /// Validating vault upgrade prerequisites.
+        /// </summary>
+        /// <returns>CheckPrerequisites response.</returns> 
+        private CheckVaultUpgradePrerequisitesResponse ValidateVaultUpgradePrerequisites()
+        {
+            CheckVaultUpgradePrerequisitesResponse response =
+                CheckVaultUpgradePrerequisitesResponse.Succeeded;
+
+            this.WriteResponse(Properties.Resources.StartingPrerequisitesCheck);
+
+            try
+            {
+                this.RecoveryServicesClient.TestVaultUpgradePrerequistes(
+                    this.VaultName,
+                    this.Location,
+                    this.ResourceType,
+                    this.TargetResourceGroupName,
+                    this.Profile.Context.Subscription.Id.ToString());
+
+                this.WriteObject(Properties.Resources.CheckPrereqSucceded);
+            }
+            catch (Exception exception)
+            {
+                this.WriteResponse(Properties.Resources.CheckPrereqFailed);
+                ExceptionDetails details =
+                    this.RecoveryServicesClient.HandleVaultUpgradeException(exception);
+                if (!string.IsNullOrEmpty(details.WarningDetails))
+                {
+                    this.WriteWarning(details.WarningDetails);
+                }
+
+                if (!string.IsNullOrEmpty(details.ErrorDetails))
+                {
+                    throw new InvalidOperationException(
```

Comment:
@neetus15 consider using WriteError(new InvalidOperationException()), instead of throwing. You get two advantages, your ErrorCategory can align with your Exception (in this case the ErrorCategory.InvalidOperation)
And users who are using your cmdlet in a script have set $ErrorView="CategoryView"
e.g.
WriteError(new ErrorRecord(ex, string.Empty, ErrorCategory.InvalidOperation, null));